hilti/runtime/include/logging.h

```diff
@@ -26,7 +26,17 @@ void fatalError(const std::string& msg) __attribute__((noreturn));
 /** Reports a warning. */
 void warning(const std::string& msg);
 
-#define HILTI_RT_DEBUG(stream, msg) hilti::rt::debug::print(stream, msg);
+/**
+ * Prints a string, or a runtime value, to a specific debug stream. This is a
+ * macro wrapper around `debug::detail::print(*)` that avoids evaluation of
+ * the arguments if nothing is going to get logged.
+ */
+#define HILTI_RT_DEBUG(stream, msg)                                                                                    \
+    {                                                                                                                  \
+        if ( ::hilti::rt::detail::globalState()->debug_logger &&                                                       \
```

Comment:
Good catch.

Reply:
This seems like a pretty bad problem so I'll backport it as well.